server/src/main/java/org/elasticsearch/action/get/TransportShardMultiGetAction.java

```diff
@@ -226,7 +231,12 @@ public void onClusterServiceClose() {
 
                     @Override
                     public void onTimeout(TimeValue timeout) {
-                        l.onFailure(new ElasticsearchException("Timed out retrying mget_from_translog[shard]", cause));
+                        if (cause instanceof AlreadyClosedException) {
+                            // Do an additional retry just in case AlreadyClosedException didn't generate a cluster update
```

Comment:
Can we add a reference to ES-10826 here?

Reply:
Added in 01b81bb6a71f1a26caf1a7c973e1f4ed22538b2d